File: Data Structures & Algorithms/merge-intervals/submission-0.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort()
        n=len(intervals)
        last=[]
        last.append(intervals[0])
        for i in range(1,n):
            curr=intervals[i]
            if last[-1][1] >= curr[0] :
                last[-1][1] = max(last[-1][1],curr[1])
            else :
                last.append(curr)
        
        return last
        
        

File: Data Structures & Algorithms/best-time-to-buy-and-sell-stock-ii/submission-0.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n=prices.size() ;
        int max_profit=0;
        int curr=prices[0] ;
        int ans=0;
        for(int i=1;i<=n;i++){
            if(i==n){
                ans+=max_profit ;
            }
            else if(prices[i]<=prices[i-1]){
                curr=prices[i] ;
                ans+=max_profit;
                max_profit=0;
            } 
            else {
                max_profit=max(max_profit,prices[i]-curr) ;
            }
        }
        return ans;
    }
};